services/api/src/resources/audit/resolvers.ts

```diff
@@ -12,69 +12,89 @@ export const getAuditLogs: ResolverFn = async (
-  }
+  // default results limit to 100
+  let limit = 100;
 
-  if (input && input.emailAddress) {
-    queryBuilder = queryBuilder.where('email_address', input.emailAddress);
-  }
+  if (input) {
+    if (input.startDate) {
+      queryBuilder = queryBuilder.where('created', '>=', input.startDate);
+    }
 
-  if (input && input.resourceId) {
-    queryBuilder = queryBuilder.where('resource_id', input.resourceId);
-  }
+    if (input.endDate) {
+      queryBuilder = queryBuilder.where('created', '<=', input.endDate);
+    }
 
-  if (input && input.resourceType) {
-    queryBuilder = queryBuilder.where('resource_type', input.resourceType);
-  }
+    if (input.startDate && input.endDate) {
+      // if a date range is requested, don't limit the results
+      limit = 0;
+    }
 
-  if (input && input.resourceDetails) {
-    queryBuilder = queryBuilder.where('resource_details','LIKE',`%${input.resourceDetails}%`);
-  }
+    // limit could be 0
+    if (input.limit != undefined) {
+      limit = input.limit;
+    }
 
-  if (input && input.linkedResourceId) {
-    queryBuilder = queryBuilder.where('linked_resource_id', input.linkedResourceId);
-  }
+    if (input.userId) {
+      queryBuilder = queryBuilder.where('usid', input.userId);
+    }
 
-  if (input && input.linkedResourceType) {
-    queryBuilder = queryBuilder.where('linked_resource_type', input.linkedResourceType);
-  }
+    if (input.emailAddress) {
+      queryBuilder = queryBuilder.where('email_address', input.emailAddress);
+    }
 
-  if (input && input.linkedResourceDetails) {
-    const l =
-    queryBuilder = queryBuilder.where('linked_resource_details','LIKE',`%${input.linkedResourceDetails}%`);
-  }
+    if (input.resourceId) {
+      queryBuilder = queryBuilder.where('resource_id', input.resourceId);
+    }
 
-  if (input && input.auditEvent) {
-    queryBuilder = queryBuilder.where('audit_event', input.auditEvent);
-  }
+    if (input.resourceType) {
+      queryBuilder = queryBuilder.where('resource_type', input.resourceType);
+    }
 
-  if (input && input.impersonatorId) {
-    queryBuilder = queryBuilder.where('impersonator_id', input.impersonatorId);
-  }
+    if (input.resourceDetails) {
+      queryBuilder = queryBuilder.where('resource_details','LIKE',`%${input.resourceDetails}%`);
+    }
 
-  if (input && input.impersonatorUsername) {
-    queryBuilder = queryBuilder.where('impersonator_username', input.impersonatorUsername);
-  }
+    if (input.linkedResourceId) {
+      queryBuilder = queryBuilder.where('linked_resource_id', input.linkedResourceId);
+    }
 
-  if (input && input.ipAddress) {
-    queryBuilder = queryBuilder.where('ip_address', input.ipAddress);
-  }
+    if (input.linkedResourceType) {
+      queryBuilder = queryBuilder.where('linked_resource_type', input.linkedResourceType);
+    }
 
-  if (input && input.source) {
-    queryBuilder = queryBuilder.where('source', input.source);
-  }
+    if (input.linkedResourceDetails) {
+      const l =
```

Comment:
Does this `l` do anything (I don't think I saw it the first time round)

Reply:
nothing to see here :hide-the-pain: